src/libraries/JANA/JApplication.h

```diff
@@ -113,6 +113,9 @@ class JApplication {
     template <typename T>
     JParameter* SetDefaultParameter(std::string name, T& val, std::string description="");
 
+    template <typename T>
+    T RegisterParameter(std::string name, const T default_val, std::string description="");
```

Comment:
Make it a reference or remove `const`.

Reply:
Shouldn't it be a const ref? The argument is that this would let the caller know we will not modify default_val, but still allow us to pass in an rvalue. We also have to remember that T is not necessary cheap-to-copy: Offhand I don't see a way to avoid at least one copy as long as we delegate to SetDefaultParameter, but I suspect that currently it makes two copies.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in

Reply:
Maybe I'm missing something, but this can't be a reference if we want the user to pass in a raw value. e.g.
~~~
int myvar = japp->RegisterParameter("myvar", 22);
~~~

Reply:
Reference to a temporary is not an issue https://godbolt.org/z/f44eT1evE